### DISCUSSION TITLE
3rd party maven dependency in react-native npm module

### DISCUSSION THREAD

**Discussion by DanielFrTB** [callstack/react-native-builder-bob#291] [Q&A]
Hi, 
I am trying to use 3rd party SDK in my react-native android module.

and I have got the same problem here:
[https://github.com/callstack/react-native-builder-bob/issues/63](url)

It seems like the module can't find the SDK and looking in the wrong place.

it only finds the SDK if I specified the in the 
`maven {
    url "xxxxx"

 }`

inside the Example-app build.gradle, the problem is that I am spcified is twice, first in the module build.gradle and the second time inside Example-app build.gradle, is it the right approach?

**Comment by rajeevverma076** [comment]
If I need to add @react-native-community/datetimepicker in my project. how I will do. If I'm adding to peerDependencies it is not calling import DateTimePicker from '@react-native-community/datetimepicker'; it is throwing an error. 
kindly suggest

**Reply by atlj** [reply]
Hey @rajeevverma076, I think the issue you are referring to is different than the original issue. Whenever you add a dependency to the `peerDependencies`, it doesn't actually get installed at user's end. It's just for warning users that they don't have a library installed while another library depends on it. You can see [the official documentation for `peerDependencies`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies). This means the end user still needs to install the library in the `peerDependencies` manually.

**Comment by eduardoborges** [comment]
Hey @DanielFrTB you have some progress with this issue?

**Comment by atlj** [comment]
You probably don't need to edit the example app's gradle at all. At `android/build.gradle`, you can find your library's build gradle file and add the dependency there. It will be enough to make the SDK usable on your library code.
